### PR TITLE
Mdast button

### PR DESCRIPTION
**Pull request by annatraussnig**
_Still needs some testing_

- Extend button component to allow using it with links
Screenshot 2019-10-17 at 10 39 37

- Setup mdast schema to parse a markdown buttonlink and match it to the button component (available for article and newsletter)
- Create a table-based button component to use in the email rendering of the markdown

### Article
article

### Newsletter email
newsletter email

### Newsletter web
newsletter web

### CODE REVIEW

File: src/templates/EditorialNewsletter/schema.js [hunk 1]
```diff
@@ -250,6 +251,32 @@ const createNewsletterSchema = ({
...
+                  matchMdast: matchParagraph,
+                  component: ({ children }) => children,
+                  rules: [{
+                    matchMdast: matchType('link'),
+                    component: ({ children }) => children,
+                    rules: globalInlines
+                  }]
+                }),
+                editorModule: 'button'
+              },
```
Comment (annatraussnig) [inline src/templates/EditorialNewsletter/schema.js]: This piece of configuration is so similar to the one in the article schema, I am still in doubt regarding whether it  would make sense to have a setButtonSchema function (or similar) somewhere that takes the component as argument and use that instead.

File: src/templates/EditorialNewsletter/email/Button.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                </td>
+              </tr>
+              </tbody>
+            </table>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  )
+}
```
Comment (annatraussnig) [inline src/templates/EditorialNewsletter/email/Button.js]: This needs a careful look…

File: src/templates/EditorialNewsletter/email/Button.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+
+export const Button = ({ href, title, primary, children, attributes }) => {
+  const styles = {
+    fontSize: '20px',
+    fontFamily: 'Helvetica, Arial, sans-serif',
```
Comment (tpreusse) [inline src/templates/EditorialNewsletter/email/Button.js]:
```suggestion
    ...fontStyles.sansSerifRegular,
```

File: src/templates/EditorialNewsletter/email/Button.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            <table
+              border="0"
+              cellSpacing="0"
+              cellPadding="0"
+              style={{borderSpacing: "0 5px", borderCollapse: "separate"}}>
+              <tbody>
+              <tr>
+                <td
+                  align="center"
+                  bgcolor={primary ? '#3CAD00' : '#4B6359'}>
```
Comment (tpreusse) [inline src/templates/EditorialNewsletter/email/Button.js]:
```suggestion
                  bgcolor={primary ? colors.primary : colors.secondary}>
```

File: src/templates/EditorialNewsletter/email/Button.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    padding: '15px 30px 15px 30px',
+    display: 'inline-block',
+  }
+
+  const primaryStyles = {
+    color: '#ffffff'
+  }
+
+  const secondaryStyles = {
+    color: '#4B6359',
```
Comment (tpreusse) [inline src/templates/EditorialNewsletter/email/Button.js]:
```suggestion
    color: colors.secondary,
```

File: src/templates/EditorialNewsletter/email/Button.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    backgroundColor: '#ffffff',
+    border: '1px solid #4B6359',
```
Comment (tpreusse) [inline src/templates/EditorialNewsletter/email/Button.js]:
```suggestion
    border: `1px solid ${colors.secondary}`,
```

File: src/templates/EditorialNewsletter/email/Button.js (hunk 3 above)
Comment (tpreusse) [reply src/templates/EditorialNewsletter/email/Button.js]:
```
import { fontStyles } from '../../../theme/fonts'
```

File: src/templates/EditorialNewsletter/email/Button.js (hunk 5 above)
Comment (tpreusse) [reply src/templates/EditorialNewsletter/email/Button.js]:
```
import colors from '../../../theme/colors'
```

**Pull request merged**